### PR TITLE
SFM-443: Impl global context in parser

### PR DESCRIPTION
**Pull request by ddold**
https://cloudbees.atlassian.net/browse/SFM-443

### CODE REVIEW

File: core/entities/variant.go [hunk 1]
```diff
@@ -81,7 +76,7 @@ func (v *variant) GetValue(ctx context.Context) string {
 
 func (v *variant) InternalGetValue(ctx context.Context) (returnValue string, isDefault bool) {
 	returnValue, isDefault = v.defaultValue, true
-	mergedContext := context.NewMergedContext(v.globalContext, ctx)
+	mergedContext := context.NewMergedContext(v.parser.GetGlobalContext(), ctx)
```
Comment (AsafRollout) [inline core/entities/variant.go:79]: from the next line it seems parser might be nil here..

Comment (ddold) [reply core/entities/variant.go]: Good point

File: core/entities/variant.go [hunk 2]
```diff
@@ -81,7 +76,13 @@ func (v *variant) GetValue(ctx context.Context) string {
 
 func (v *variant) InternalGetValue(ctx context.Context) (returnValue string, isDefault bool) {
 	returnValue, isDefault = v.defaultValue, true
-	mergedContext := context.NewMergedContext(v.globalContext, ctx)
+	var mergedContext context.Context = nil
+
+	if v.parser != nil {
+		mergedContext = context.NewMergedContext(v.parser.GetGlobalContext(), ctx)
+	} else {
+		mergedContext = context.NewMergedContext(nil, ctx)
```
Comment (AsafRollout) [inline core/entities/variant.go:84]: won't context.NewMergedContext(nil, ctx) be the same as just using ctx ?
so
var mergedContext context.Context = ctx
if v.parser != nil { mergedContext = context.NewMergedC... }
?

**Review by AsafRollout — APPROVED** [review]
please add tests if possible (evaluation using global context, and local context)

**Pull request merged**